tensorboard/writer/event_file_writer.py

```diff
@@ -0,0 +1,231 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Writes events to disk in a logdir."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os.path
+import socket
+import threading
+import time
+
+import six
+
+from ..proto import event_pb2
+from .record_writer import RecordWriter, directory_check
+
+
+class EventsWriter(object):
+    '''Writes `Event` protocol buffers to an event file.'''
+
+    def __init__(self, file_prefix, filename_suffix=''):
+        '''
+        Events files have a name of the form
+        '/some/file/path/events.out.tfevents.[timestamp].[hostname]'
+        '''
+        self._file_name = file_prefix + ".out.tfevents." + str(time.time())[:10] + "." +\
+            socket.gethostname() + filename_suffix
+
+        self._num_outstanding_events = 0
+
+        self._py_recordio_writer = RecordWriter(self._file_name)
+
+        # Initialize an event instance.
+        self._event = event_pb2.Event()
+
+        self._event.wall_time = time.time()
+
+        self._lock = threading.Lock()
+
+        self.write_event(self._event)
+
+    def write_event(self, event):
+        '''Append "event" to the file.'''
+
+        # Check if event is of type event_pb2.Event proto.
+        if not isinstance(event, event_pb2.Event):
+            raise TypeError("Expected an event_pb2.Event proto, "
+                            " but got %s" % type(event))
+        return self._write_serialized_event(event.SerializeToString())
+
+    def _write_serialized_event(self, event_str):
+        with self._lock:
+            self._num_outstanding_events += 1
+            self._py_recordio_writer.write(event_str)
+
+    def flush(self):
+        '''Flushes the event file to disk.'''
+        with self._lock:
+            self._num_outstanding_events = 0
+            self._py_recordio_writer.flush()
+        return True
+
+    def close(self):
+        '''Call self.flush().'''
+        return_value = self.flush()
+        with self._lock:
+            self._py_recordio_writer.close()
+        return return_value
+
+
+class EventFileWriter(object):
+    """Writes `Event` protocol buffers to an event file.
+    The `EventFileWriter` class creates an event file in the specified directory,
+    and asynchronously writes Event protocol buffers to the file. The Event file
+    is encoded using the tfrecord format, which is similar to RecordIO.
+    @@__init__
```

Comment:
Can we match the rest of TensorBoard for this docstring? I think you can actually just remove them but @nfelt might have a stronger opinion.

Reply:
Retaining the docstrings seems useful, but yes, please follow the rest of TensorBoard in style.